orc8r/gateway/python/magma/magmad/config_manager.py

```diff
@@ -145,4 +146,29 @@ def did_mconfig_change(serv_name):
             if srv in mconfig.configs_by_key:
                 configs_by_key[srv] = mconfig.configs_by_key.get(srv)
 
+        # version should be in 1.1.1 format with only positive numbers allowed
+        VERSION_REGEX = re.compile(r"[0-9]+\.(?P<minor_version>[0-9]+)\.[0-9]+")
+
+        agw_version = self._magmad_service.version
+        unpacked_mconfig = mconfig.configs_by_key.get(MAGMAD)
```

Comment:
Could we maybe add a few sentences on what these are checking?  And let's restructure the if conditions so that we don't have too many nested ones.

Reply:
i am not sure if i see the nested condition, all if conditions looks to be 1 level deep. Arent they?

Reply:
It's not that it's overly nested but there's just a lot of conditions with some nests.  It would be helpful for future readers if there's a summary of what it does and what it's checks for. 

It would be good to pull this logic into a separate function, and a comment or two can't hurt. My preference is to always check the negative case (or the case that involves less logic) so that functions can return early, but that decision is up to you.

Reply:
moved the checks to a different function with more helpful comments :)